### PR TITLE
Make OStream#write accept multiple arguments

### PR DESCRIPTION
**Pull request by kojix2**
#205
image

### CODE REVIEW

File: lib/iruby/ostream.rb [hunk 1]
```diff
@@ -25,9 +25,10 @@ def read(*args)
     alias_method :next, :read
     alias_method :readline, :read
 
-    def write(s)
+    def write(*args)
+      str = args.join
```
Comment (mrkn) [inline lib/iruby/ostream.rb]: I think the following code is better than using `Array#join` for mimicking `IO#write`'s behavior.

```
str = StringIO.open {|sio| sio.write(*args); sio.string }
```

Comment (mrkn) [reply lib/iruby/ostream.rb]: @kojix2 Could you please rewrite also puts and printf by using StringIO?

Comment (kojix2) [reply lib/iruby/ostream.rb]: I'll try.

Comment (kojix2) [reply lib/iruby/ostream.rb]: What should I do next?

File: lib/iruby/ostream.rb [hunk 2]
```diff
@@ -25,26 +25,33 @@ def read(*args)
     alias_method :next, :read
     alias_method :readline, :read
 
-    def write(s)
-      raise 'I/O operation on closed file' unless @session
-      @session.send(:publish, :stream, name: @name, text: s.to_s)
-      nil
+    def write(*obj)
+      str = StringIO.open {|sio| sio.write(*obj); sio.string }
```
Comment (mrkn) [inline lib/iruby/ostream.rb]: We can refactor this line by introducing new private method like:

```ruby
def write(*obj)
  str = build_string {|sio| sio.write(*obj) }
end

private

def build_string
  StringIO.open {|sio| yield(sio); sio.string }
end
```

File: lib/iruby/ostream.rb [hunk 3]
```diff
@@ -25,26 +25,33 @@ def read(*args)
...
+      session_send(str)
     end
     alias_method :<<, :write
     alias_method :print, :write
 
-    def printf(*fmt)
-      write sprintf(*fmt)
+    def printf(format, *obj)
+      str = StringIO.open {|sio| sio.printf(format, *obj); sio.string }
```
Comment (mrkn) [inline lib/iruby/ostream.rb]: same as above

File: lib/iruby/ostream.rb [hunk 4]
```diff
@@ -25,26 +25,33 @@ def read(*args)
...
+      session_send(str)
     end
 
-    def puts(*lines)
-      lines = [''] if lines.empty?
-      lines.each { |s| write("#{s}\n")}
-      nil
+    def puts(*obj)
+      str = StringIO.open {|sio| sio.puts(*obj); sio.string }
```
Comment (mrkn) [inline lib/iruby/ostream.rb]: same as above

File: lib/iruby/ostream.rb (hunk 2 above)
Comment (kojix2) [reply lib/iruby/ostream.rb]: Thanks. Done.

File: lib/iruby/ostream.rb [hunk 5]
```diff
@@ -19,26 +19,26 @@ def isatty
     end
     alias_method :tty?, :isatty
 
-    def read(*args)
+    def read(*_args)
```
Comment (mrkn) [inline lib/iruby/ostream.rb]: Why do you rename this?

File: lib/iruby/ostream.rb [hunk 6]
```diff
@@ -19,32 +19,44 @@ def isatty
...
+      session_send(str)
     end
 
     def writelines(lines)
       lines.each { |s| write(s) }
     end
+
+    private
+
+    def building_string
```
Comment (mrkn) [inline lib/iruby/ostream.rb]: I think a gerund is not fit to the method name.

File: lib/iruby/ostream.rb (hunk 5 above)
Comment (kojix2) [reply lib/iruby/ostream.rb]: argument is not used.

File: lib/iruby/ostream.rb (hunk 6 above)
Comment (kojix2) [reply lib/iruby/ostream.rb]: Hmm. That makes sense..

File: lib/iruby/ostream.rb (hunk 5 above)
Comment (mrkn) [reply lib/iruby/ostream.rb]: Ah, I got it.

Comment (kojix2) [reply lib/iruby/ostream.rb]: Okay. I'll put it back.

**Pull request merged**